pkg/api/types.go

```diff
@@ -1281,3 +1281,36 @@ func (k *KubernetesConfig) SetCloudProviderRateLimitDefaults() {
 		k.CloudProviderRateLimitBucket = DefaultKubernetesCloudProviderRateLimitBucket
 	}
 }
+
+//GetCloudSpecConfig returns the Kubernetes container images URL configurations based on the deploy target environment.
+//for example: if the target is the public azure, then the default container image url should be k8s.gcr.io/...
+//if the target is azure china, then the default container image should be mirror.azure.cn:5000/google_container/...
+func (cs *ContainerService) GetCloudSpecConfig() AzureEnvironmentSpecConfig {
+	switch getCloudTargetEnv(cs.Location) {
+	case azureChinaCloud:
+		return AzureChinaCloudSpec
+	case azureGermanCloud:
+		return AzureGermanCloudSpec
+	case azureUSGovernmentCloud:
+		return AzureUSGovernmentCloud
+	default:
+		return AzureCloudSpec
+	}
+}
+
+// getCloudTargetEnv determines and returns whether the region is a sovereign cloud which
+// have their own data compliance regulations (China/Germany/USGov) or standard
+//  Azure public cloud
+func getCloudTargetEnv(location string) string {
```

Comment:
Should this be a method of `ContainerService`?

Reply:
Oh yes. That would be more consistent

Reply:
Fixed